geomstats/geometry/discrete_curves.py

```diff
@@ -275,6 +283,54 @@ def random_point(self, n_samples=1, bound=1.0):
         )(sampling_times)
         return self.projection(point)
 
+    def interpolate(self, point):
+        """Interpolate between the sampling points of a discrete curve.
+
+        Parameters
+        ----------
+        point : array-like, shape=[..., k_sampling_points, ambient_dim]
+            Discrete curve starting at the origin.
+
+        Returns
+        -------
+        spline : function
+            Cubic spline that interpolates between the sampling points
+            of the discrete curve.
+        """
+        k_sampling_points = self.k_sampling_points
+        t_space = gs.linspace(0.0, 1.0, k_sampling_points)
+        point_with_origin = insert_zeros(point, axis=-self.point_ndim)
+        is_batch = check_is_batch(self.point_ndim, point)
+        if not is_batch:
+            return CubicSpline(t_space, point_with_origin, axis=-self.point_ndim)
+        return [
+            CubicSpline(t_space, point_with_origin_, axis=-self.point_ndim)
+            for point_with_origin_ in point_with_origin
+        ]
+
+    def length(self, point):
+        """Compute the length of a discrete curve.
+
+        This is the integral of the absolute value of the velocity.
+
+        Parameters
+        ----------
+        point : array-like, shape=[..., k_sampling_points, ambient_dim]
+            Discrete curve starting at the origin.
+
+        Returns
+        -------
+        length : array-like, shape=[..., ]
+            Length of the discrete curve.
+        """
+        point_with_origin = self.insert_origin(point)
+        velocity = forward_difference(point_with_origin, axis=-self.point_ndim)
+        return L2CurvesMetric(self).norm(velocity, point_with_origin[..., :-1, :])
```

Comment:
maybe we should instantiate this in `__init__` (e.g. `self.l2_curves_metric`). first I thought to make it private, but it seems to be used often below in the aligners. By putting there we can do something like `total_space.l2_curves_metric`, avoiding instantiating every time.

another option, which is probably more in accordance with geomstats design, is to do

```python
self.discrete_curves_with_l2 = self.new(equip=False).equip_with_metric(
    L2CurvesMetric,
)
```
(names are indicative)

I would be more comfortable with this later approach, because it will consist of having two Riemannian manifolds `(M, g)`. I think the fact one is used within the other is not important, but we are trying to avoid having something like `(M, g1, g2)`.